tasty-query/shared/src/test/scala/tastyquery/SubtypingSuite.scala

```diff
@@ -794,6 +794,35 @@ class SubtypingSuite extends UnrestrictedUnpicklingSuite:
     assertNeitherSubtype(ConcreteSimplePathsChildClass.appliedRef, refineTerm("abstractTerm", listOf(defn.BooleanType)))
       .withRef[ConcreteSimplePathsChild, SimplePaths { def abstractTerm: List[Boolean] }]
 
+    assertEquiv(
+      refineTerm("abstractTerm", defn.StringType, isStable = true),
+      refineTerm("abstractTerm", defn.StringType, isStable = true)
+    )
+      .withRef[SimplePaths { val abstractTerm: String }, SimplePaths { val abstractTerm: String }]
+    assertStrictSubtype(refineTerm("abstractTerm", defn.StringType, isStable = true), SimplePathsClass.appliedRef)
+      .withRef[SimplePaths { val abstractTerm: String }, SimplePaths]
+
+    // no withRef on the following because dotc does not take stability into account, apparently
+    assertStrictSubtype(
```

Comment:
if there is a difference to dotc isn't that incorrect?

Reply:
I mean, yes, dotc is incorrect. :laughing: It's unsound. Scala 2 is right:
```scala
$ cs launch scala:2.13.11
Welcome to Scala 2.13.11 (OpenJDK 64-Bit Server VM, Java 1.8.0_372).
Type in expressions for evaluation. Or try :help.

scala> def foo(x: Any { def foo: Int }): Any { val foo: Int } = x
                                                                ^
       error: type mismatch;
        found   : Any{def foo: Int}
        required: Any{val foo: Int}

scala> 
$ cs launch scala:3.3.0
Welcome to Scala 3.3.0 (1.8.0_372, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                    
scala> def foo(x: Any { def foo: Int }): Any { val foo: Int } = x
def foo(x: Any{def foo: Int}): Any{val foo: Int}
                                                                                                    
scala> 
```

Reply:
I reported https://github.com/lampepfl/dotty/issues/18047